audit/hashstructure.go

```diff
@@ -380,6 +381,9 @@ func (w *hashWalker) Primitive(v reflect.Value) error {
 		si := int(w.csKey[len(w.cs)-1].Int())
 		s.Slice(si, si+1).Index(0).Set(resultVal)
 	default:
+		if !setV.CanSet() {
+			return nil
+		}
```

Comment:
Couldn't quite figure out how to make it settable. If anyone has ideas, I'm open to it! But for now, I figure that returning early is better than a panic.

Reply:
Let's not merge this PR please until we are done with the thread on Slack.

Maybe we could return an error for now (to make sure we don't panic) and spend some time discussing a longer therm solution. I can see our users getting pretty upset about data that they expected to be HMAC'd turning up in the log in plaintext. Maybe it's fine and I am overreacting, though 😄.